spec/mqtt/integrations/connect_spec.cr

```diff
@@ -0,0 +1,286 @@
+require "../spec_helper"
+
+module MqttSpecs
+  extend MqttHelpers
+  extend MqttMatchers
+  describe "connect [MQTT-3.1.4-1]" do
+    describe "when client already connected" do
+      it "should replace the already connected client [MQTT-3.1.4-2]" do
+        with_server do |server|
+          with_client_io(server) do |io|
+            connect(io)
+            with_client_io(server) do |io2|
+              connect(io2)
+              io.should be_closed
+            end
+          end
+        end
+      end
+    end
+
+    describe "receives connack" do
+      describe "with expected flags set" do
+        it "no session present when reconnecting a non-clean session with a clean session [MQTT-3.1.2-6]" do
+          with_server do |server|
+            with_client_io(server) do |io|
+              connect(io, clean_session: false)
+
+              # LavinMQ won't save sessions without subscriptions
+              subscribe(io,
+                topic_filters: [subtopic("a/topic", 0u8)],
+                packet_id: 1u16
+              )
+              disconnect(io)
+              sleep 0.1
```

Comment:
We should investigate why this is needed. It may be a bug in the connect code.

Reply:
#803 will probably fix this

Reply:
oh nice, i'll try it out!

Reply:
I'm not sure I'm done with that fix yet, though. Maybe.